### ISSUE TITLE
torch.norm will give a scalar, Is this a bug?

### ISSUE THREAD

**Issue by fanlu** [HarryVolek/PyTorch_Speaker_Verification#16]
https://github.com/HarryVolek/PyTorch_Speaker_Verification/blob/29232c8e3512b576b8c60b7b5aacea578c1e1369/speech_embedder_net.py#L32
torch.norm will give a scalar, and In paper eq(4), I think x should be normalized by each embedding.
F.normalize(x, dim=-1, eps=1e-6), Am I right?

**Comment by HarryVolek** [comment]
Torch.norm will give a scalar, as expected.
